### ISSUE TITLE
Acer Chromebook  317 No Audio, Intel N4500

### ISSUE THREAD

**Issue by xSPARExSTEWx** [cb-linux/breath#167]
Just tried the new Easter release as previous versions would not boot. It boots now on Jasper Lake CPUs, but audio is not for me at least.

Tried all the steps and they did not work. If I do the "VERSION=ALT bash updatekernel.sh" part it makes it not bootable anymore.

Used Ubuntu 21.04 and 21.10 kde and budgie.

lsmod output:
[lsmod.txt](https://github.com/cb-linux/breath/files/8527203/lsmod.txt)

find /usr/share/alsa output:
[find usr_share_alsa.txt](https://github.com/cb-linux/breath/files/8527208/find.usr_share_alsa.txt)

sudo dmesg output:
[dmesg.txt](https://github.com/cb-linux/breath/files/8527227/dmesg.txt)

sudo alsa-info does not work. Do I do this in Breath?

CB317-1H-C994
N21Q4
MAGPIE-TQAU

**Comment by MilkyDeveloper** [comment]
Yeah, you want to run all of these commands in Breath. The alt kernel is only meant to be used for KBL/Skylake devices.

**Comment by MilkyDeveloper** [comment]
Also, it looks like you have a Realtek device, so you would use `SOUNDCARD=rtk sof-setup-audio`

**Comment by xSPARExSTEWx** [comment]
> 

I have tried all the commands and none of them work.

Only audio device it shows is "Dummy Output"

Here are those commands done in Breath after I did `SOUNDCARD=rtk sof-setup-audio`

lsmod output:
[lsmod.txt](https://github.com/cb-linux/breath/files/8536866/lsmod.txt)

find /usr/share/alsa output:
[find alsa.txt](https://github.com/cb-linux/breath/files/8536867/find.alsa.txt)

sudo dmesg output:
[dmesg.txt](https://github.com/cb-linux/breath/files/8536868/dmesg.txt)

sudo alsa-info output:
http://alsa-project.org/db/?f=1aa2a6bb81d2c4bde22c23d5ff83f553dd390a8

**Comment by MilkyDeveloper** [comment]
Seems like our primary issue is:

> [   19.659614] sof-audio-pci-intel-icl 0000:00:1f.3: error: cl_copy_fw: timeout HDA_DSP_SRAM_REG_ROM_STATUS read
[   19.660120] sof-audio-pci-intel-icl 0000:00:1f.3: status: manifest loaded - code 00000003
[   19.660148] sof-audio-pci-intel-icl 0000:00:1f.3: error: extended rom status:  0x3 0x0 0x0 0x0 0x0 0x0 0x2100186 0x0
[   19.660150] sof-audio-pci-intel-icl 0000:00:1f.3: error: load fw failed ret: -110
[   19.660201] sof-audio-pci-intel-icl 0000:00:1f.3: error: failed to reset DSP
[   19.660203] sof-audio-pci-intel-icl 0000:00:1f.3: error: failed to boot DSP firmware -110
[   19.710636] sof-audio-pci-intel-icl 0000:00:1f.3: error: hda_dsp_core_reset_enter: timeout on HDA_DSP_REG_ADSPCS read
[   19.710641] sof-audio-pci-intel-icl 0000:00:1f.3: error: dsp core reset failed: core_mask 1
[   19.710859] sof-audio-pci-intel-icl 0000:00:1f.3: error: sof_probe_work failed err: -110

**Comment by MilkyDeveloper** [comment]
I know you’ve already ran this, but, just for confirmation, I believe using firmware from ChromeOS can fix this. Run `setup-audio`, tell me if there’s any errors, and then do the rtk sof-setup-audio.

**Comment by xSPARExSTEWx** [comment]
Looks like a few errors come up when I run both. Here are the outputs for both commands.

[setup-audio.txt](https://github.com/cb-linux/breath/files/8543265/setup-audio.txt)

[SOUNDCARD=rtk sof-setup-audio.txt](https://github.com/cb-linux/breath/files/8543267/SOUNDCARD.rtk.sof-setup-audio.txt)

**Comment by MilkyDeveloper** [comment]
Looks like your firmware is copied, so no worries in `setup-audio`. What I'm confused about is
```
Failed to connect to bus: $DBUS_SESSION_BUS_ADDRESS and $XDG_RUNTIME_DIR not defined (consider using --machine=<user>@.host --user to connect to bus of other user)
```

Are you running this within a chroot? I think both of these variables should always be defined, especially when using a desktop.

**Comment by MilkyDeveloper** [comment]
What about running this from Breath:
```
git clone https://github.com/thesofproject/sof-bin —depth 1
sudo ./install.sh v2.1.x
```

**Comment by xSPARExSTEWx** [comment]
`git clone https://github.com/thesofproject/sof-bin —depth 1` Does not work, but I did `git clone --depth 1 https://github.com/thesofproject/sof-bin`

When I run `sudo ./install.sh v2.1.x` I get the following

> "+ ln -sT sof-v2.1.x /lib/firmware/intel/sof"

> "ln: failed to create symbolic link '/lib/firmware/intel/sof': File exists"

> "+ set +x"

> "./install.sh ERROR: /lib/firmware/intel/sof already installed? (Re)move it first."

Since it is already installed I ran the scripts to upgrade it that sof-dev says to do.
`sudo mv /lib/firmware/intel/sof* /home/matt/sof-bin/backup`
`sudo mv /usr/local/bin/sof-* /home/matt/sof-bin/backup`

I get the following error after doing that

> "+ ln -sT sof-v2.1.x /lib/firmware/intel/sof"

> "+ ln -sT sof-tplg-v2.1.x /lib/firmware/intel/sof-tplg"

> "+ rsync -a ./sof*v2.1.x /lib/firmware/intel/"

> "rsync: [sender] link_stat "/home/matt/sof-bin/sof*v2.1.x" failed: No such file or directory (2)"

> "rsync error: some files/attrs were not transferred (see previous errors) (code 23) at main.c(1330) [sender=3.2.3]"

**Comment by MilkyDeveloper** [comment]
```
cd v2.1.x
rsync -a sof*v2.1.1   /lib/firmware/intel/
ln -s sof-v2.1.1      /lib/firmware/intel/sof
ln -s sof-tplg-v2.1.1 /lib/firmware/intel/sof-tplg
rsync tools-v2.1.1/*  /usr/local/bin
```

**Comment by xSPARExSTEWx** [comment]
I am able to run those commands if I move the files again, but after running them and restarting still no audio.

**Comment by MilkyDeveloper** [comment]
What about uncompressing: [sof-jsl-signed.ri.gz](https://github.com/cb-linux/breath/files/8558143/sof-jsl-signed.ri.gz) and copying it to `/lib/firmware/intel/sof/sof-jsl.ri` (this is a file)?

Post another `sudo alsa-info`.

**Comment by xSPARExSTEWx** [comment]
After doing that it still did not work.

I tried what we did above again and that did not work either
http://alsa-project.org/db/?f=6e046aa64b62cc12dbf50b3b3ab2cd25b0e31f45

I tried doing the setup-audio again and then doing SOUNDCARD=rtk sof-setup-audio, but get an error `sof-setup-audio: command not found`. Going into the audio settings I no longer see "Dummy Output" it is just blank and pressing the test button it says the device is (null)
http://alsa-project.org/db/?f=9c79bbbccc8d0e6a7ae5626e305a1942370372e2

I moved the file over again as I forgot to do alsa-info before I did the other things so here it is after restarting (I have been restarting after each section)
http://alsa-project.org/db/?f=200fe3ebbf176209e8834871e0dcaa2290a27d86

**Comment by energy-x** [comment]
Is there any additional troubleshooting needed for this issue? I can try whatever necessary to try to resolve. Thanks!

**Comment by MilkyDeveloper** [comment]
@energy-x Join the [Discord](https://discord.gg/MjPKmWDr) for faster communication. Feel free to follow the instructions in the JSL Audio thread and report there.

**Comment by xSPARExSTEWx** [comment]
I have actually sold mine because it was a bit underpowered for what I needed it for.

FYI I did try the Summer Release and the audio still was not working.

**Comment by MilkyDeveloper** [comment]
We've fixed this since then.

**Comment by energy-x** [comment]
> We've fixed this since then.

I just tried Ubuntu with gnome and xfce on the MAGPIE and after running sof-setup-audio, still no audio. A device showed up in xfce after reboot, but no sound was produced. With KDE/Ubuntu I got an error at the end of the script about connecting to the local host and still had dummy interface after reboot. Which Linux versions is this fixed on?
